app/services/bulk_url_uploader.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+class BulkUrlUploader
+  MAXIMUM_FILE_SIZE = 10.megabytes
+  VALID_CONTENT_TYPES = %w[text/plain].freeze
+
+  attr_reader :results
+
+  class Error < StandardError
+  end
+
+  class Results
+    attr_accessor :searchgov_domains, :ok_count, :error_count, :name
+
+    def initialize(name)
+      @name = name
+      @ok_count = 0
+      @error_count = 0
+      @searchgov_domains = Set.new
+      @errors = Hash.new { |hash, key| hash[key] = [] }
+    end
+
+    def add_ok(url)
+      self.ok_count += 1
+      searchgov_domains << url.searchgov_domain
+    end
+
+    def add_error(error_message, url)
+      self.error_count += 1
+      @errors[error_message] << url
+    end
+
+    def total_count
+      ok_count + error_count
+    end
+
+    def error_messages
+      @errors.keys
+    end
+
+    def urls_with(error_message)
+      @errors[error_message]
+    end
+  end
+
+  class UrlFileValidator
+    def initialize(uploaded_file)
+      @uploaded_file = uploaded_file
+    end
+
+    def validate!
+      ensure_present
+      ensure_valid_content_type
+      ensure_not_too_big
+    end
+
+    def ensure_valid_content_type
+      return if BulkUrlUploader::VALID_CONTENT_TYPES.include?(@uploaded_file.content_type)
```

Comment:
We can leave this for now, but try to use accessor methods instead of instance variables